From 66761cb9443b4cc319edee9c10e2cc0f6284c970 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Wed, 24 Apr 2024 16:02:44 +0200
Subject: [PATCH 0823/1695] stm32mp2: adapt RISAF device tree bindings for
 STM32MP21

Add a new MCE encryption definition only available on STM32MP21.
This encryption mode protects against side-channel attacks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a0ad51b7403c1333b96f0bff30f2c90adf8a966b stm32mp2: adapt RISAF device tree bindings for STM32MP21]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I7ca8f9a85f0aa629eac5455bb1029731f8201e17
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376412
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/include/dt-bindings/soc/stm32mp25-risaf.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/include/dt-bindings/soc/stm32mp25-risaf.h b/core/include/dt-bindings/soc/stm32mp25-risaf.h
index 7e1d4ca3e..ac9d53420 100644
--- a/core/include/dt-bindings/soc/stm32mp25-risaf.h
+++ b/core/include/dt-bindings/soc/stm32mp25-risaf.h
@@ -15,6 +15,9 @@
 
 /* RISAF encryption modes */
 #define RIF_ENC_DIS		0x0
+#if defined(CFG_STM32MP21)
+#define RIF_ENC_MCE_EN		0x1 /* Side-channel attack protection */
+#endif /* defined(CFG_STM32MP21) */
 #define RIF_ENC_EN		0x2
 
 #define DT_RISAF_EN_SHIFT	4
-- 
2.47.2

